minos/api_gateway/rest/handler.py

```diff
@@ -98,3 +121,33 @@ async def _clone_response(response: ClientResponse) -> web.Response:
     return web.Response(
         body=await response.read(), status=response.status, reason=response.reason, headers=response.headers,
     )
+
+
+async def authenticate(
+    address: str, port: str, method: str, path: str, authorization_headers: CIMultiDict
+) -> dict[str, str]:
+    authentication_url = URL(f"http://{address}:{port}/{path}")
+    authentication_method = method
+    logger.info("Authenticating request...")
+
+    try:
+        async with ClientSession(headers=authorization_headers) as session:
+            async with session.request(method=authentication_method, url=authentication_url) as response:
+                if response.ok:
+                    return await response.json()
+                else:
+                    raise InvalidAuthenticationException
+    except ClientConnectorError:
+        raise web.HTTPServiceUnavailable(text="The requested endpoint is not available.")
```

Comment:
As the exception raised here will be catched by the `get_user` function, i think it's better to also return an `InvalidAuthenticationException` to keep the logic as simple as possible and try to keep separated the `aiohttp` status code exceptions with the application level exceptions although the `ClientConnectorError` is an `aiohttp` exception.

Reply:
Done.